compiler/noirc_evaluator/src/brillig/brillig_ir.rs

```diff
@@ -578,13 +620,13 @@ impl BrilligContext {
                 self.store_instruction(size_pointer, size_constant);
                 self.deallocate_register(size_constant);
             }
-            RegisterOrMemory::HeapArray(HeapArray { pointer, size }) => {
+            RegisterOrMemory::HeapArray(HeapArray { pointer, size, reference_count: _ }) => {
                 self.store_instruction(variable_pointer, pointer);
                 let size_constant = self.make_constant(Value::from(size));
                 self.store_instruction(size_pointer, size_constant);
                 self.deallocate_register(size_constant);
             }
-            RegisterOrMemory::HeapVector(HeapVector { pointer, size }) => {
+            RegisterOrMemory::HeapVector(HeapVector { pointer, size, reference_count: _ }) => {
```

Comment:
We also would need to update allocate_variable_instruction implementation to take into account that now variables can be up to 3 items (pointer, length and reference count)

Reply:
I don't understand these load and store methods or how I'd extend them to add a reference count. It looks like they only take in a source_variable for the array pointer, and a address_register that we use to store the size? I'm not sure why we'd store the size where the address goes, nor am I sure how this would be extended to store a reference count as well.